docs/architecture/grpc-for-wcf-developers/client-libraries.md

```diff
@@ -63,28 +61,37 @@ namespace Grpc.Core
 The `GrpcStreamSubscription` class handles the enumeration of the `IAsyncStreamReader`:
 
 ```csharp
-public class GrpcStreamSubscription : IDisposable
+public class GrpcStreamSubscription<T> : IDisposable
 {
-    private readonly Task _task;
+    private readonly IAsyncStreamReader<T> _reader;
+    private readonly IObserver<T> _observer;
+
     private readonly CancellationTokenSource _tokenSource;
+
+    private readonly Task _task;
+
     private bool _completed;
 
-    public GrpcStreamSubscription(IAsyncStreamReader<T> reader, IObserver<T> observer, CancellationToken token)
+    public GrpcStreamSubscription(IAsyncStreamReader<T> reader, IObserver<T> observer, CancellationToken token = default)
     {
-        Debug.Assert(reader != null);
-        Debug.Assert(observer != null);
+        _reader = reader ?? throw new ArgumentNullException(nameof(reader));
+        _observer = observer ?? throw new ArgumentNullException(nameof(observer));
+
         _tokenSource = new CancellationTokenSource();
         token.Register(_tokenSource.Cancel);
-        _task = Run(reader, observer, _tokenSource.Token);
+
+        _completed = false;
```

Comment:
```suggestion
```

This is the default value, and is _not_ necessary. Was it added solely for readability?

Reply:
Yes, I thought it was making the process simpler to understand instead of having to only have _completed being set to true.

But we can remove it, technically it's true that it is redundant.

Reply:
I've applied the suggestion